x-pack/plugins/security_solution/server/lib/siem_migrations/rules/data/rule_migrations_data_rules_client.ts

```diff
@@ -395,3 +407,89 @@ const conditions = {
     return [this.isFullyTranslated(), this.isNotInstalled()];
   },
 };
+
+const missing = (direction: estypes.SortOrder = 'asc') =>
+  direction === 'desc' ? '_last' : '_first';
+
+const sortingOptions = {
+  author(direction: estypes.SortOrder = 'asc'): estypes.SortCombinations {
+    return {
+      'elastic_rule.prebuilt_rule_id': {
+        order: direction,
+        nested: { path: 'elastic_rule' },
+        missing: missing(direction),
+      },
+    };
+  },
+  severity(direction: estypes.SortOrder = 'asc'): estypes.SortCombinations {
+    const field = 'elastic_rule.severity';
+    return {
+      _script: {
+        order: direction,
+        type: 'number',
+        script: {
+          source: `
+          if (doc.containsKey('${field}') && !doc['${field}'].empty) {
+            def value = doc['${field}'].value.toLowerCase();
+            if (value == 'critical') { return 3 }
+            if (value == 'high') { return 2 }
+            if (value == 'medium') { return 1 }
+            if (value == 'low') { return 0 }
+          }
+          return -1;
+          `,
+          lang: 'painless',
+        },
+        nested: { path: 'elastic_rule' },
+      },
+    };
+  },
+  status(direction: estypes.SortOrder = 'asc'): estypes.SortCombinations {
+    const field = 'translation_result';
+    return {
+      _script: {
+        order: direction,
+        type: 'number',
+        script: {
+          source: `
+          if (doc.containsKey('${field}') && !doc['${field}'].empty) {
+            def value = doc['${field}'].value.toLowerCase();
+            if (value == 'full') { return 2 }
+            if (value == 'partial') { return 1 }
+            if (value == 'untranslatable') { return 0 }
+          }
+          return -1;
+          `,
+          lang: 'painless',
+        },
+      },
+    };
+  },
+  updated(direction: estypes.SortOrder = 'asc'): estypes.SortCombinations {
+    return { updated_at: direction };
+  },
+  name(direction: estypes.SortOrder = 'asc'): estypes.SortCombinations {
+    return { 'elastic_rule.title.keyword': { order: direction, nested: { path: 'elastic_rule' } } };
+  },
+};
+
+const DEFAULT_SORTING: estypes.Sort = [
+  sortingOptions.status('desc'),
+  sortingOptions.author('desc'),
+  sortingOptions.severity(),
+  sortingOptions.updated(),
+];
+
+const getSortingOptions = (sort?: RuleMigrationSort): estypes.Sort => {
+  if (!sort?.sortField) {
+    return DEFAULT_SORTING;
+  }
+  const sortingOptionsMap: { [key: string]: estypes.Sort } = {
+    'elastic_rule.title': sortingOptions.name(sort.sortDirection),
+    'elastic_rule.severity': sortingOptions.severity(sort.sortDirection),
+    'elastic_rule.prebuilt_rule_id': sortingOptions.author(sort.sortDirection),
+    translation_result: sortingOptions.status(sort.sortDirection),
+    updated_at: sortingOptions.updated(sort.sortDirection),
+  };
+  return sortingOptionsMap[sort.sortField] ?? DEFAULT_SORTING;
+};
```

Comment:
Wouln't it be better to create the map only once?
```suggestion

const sortingOptionsMap: { [key: string]: estypes.Sort } = {
  'elastic_rule.title': sortingOptions.name,
  'elastic_rule.severity': sortingOptions.severity,
  'elastic_rule.prebuilt_rule_id': sortingOptions.author,
  translation_result: sortingOptions.status,
  updated_at: sortingOptions.updated,
};

const getSortingOptions = (sort?: RuleMigrationSort): estypes.Sort => {
  if (!sort?.sortField) {
    return DEFAULT_SORTING;
  }
  return sortingOptionsMap[sort.sortField]?.(sort.sortDirection) ?? DEFAULT_SORTING;
};
```

Reply:
done